server/src/main/java/org/opensearch/index/translog/transfer/TranslogCheckpointTransferSnapshot.java

```diff
@@ -69,6 +70,13 @@ public Set<TransferFileSnapshot> getCheckpointFileSnapshots() {
         return translogCheckpointFileInfoTupleSet.stream().map(Tuple::v2).collect(Collectors.toSet());
     }
 
+    public void close() throws IOException {
+        for (Tuple<TranslogFileSnapshot, CheckpointFileSnapshot> tlogCkpTuple : translogCheckpointFileInfoTupleSet) {
+            tlogCkpTuple.v1().close();
+            tlogCkpTuple.v2().close();
```

Comment:
What if any of these close() calls throw an exception? Is it okay not to close the rest of them?

Reply:
Good catch @andrross . We should make best effort to close all the file snapshots here. Added that handling now.